src/core/lib/security/credentials/tls/grpc_tls_credentials_options.h

```diff
@@ -292,31 +180,35 @@ struct grpc_tls_credentials_options
   void set_max_tls_version(grpc_tls_version max_tls_version) {
     max_tls_version_ = max_tls_version;
   }
-  void set_key_materials_config(
-      grpc_core::RefCountedPtr<grpc_tls_key_materials_config> config) {
-    key_materials_config_ = std::move(config);
-  }
-  void set_credential_reload_config(
-      grpc_core::RefCountedPtr<grpc_tls_credential_reload_config> config) {
-    credential_reload_config_ = std::move(config);
-  }
   void set_server_authorization_check_config(
       grpc_core::RefCountedPtr<grpc_tls_server_authorization_check_config>
           config) {
     server_authorization_check_config_ = std::move(config);
   }
 
+  void set_certificate_provider(grpc_tls_certificate_provider* provider) {
+    grpc_tls_certificate_provider_ = provider;
+  }
+
+  void set_root_cert_name(std::string root_cert_name) {
+    root_cert_name_ = std::move(root_cert_name);
+  }
+
+  void set_identity_cert_name(std::string identity_cert_name) {
+    identity_cert_name_ = std::move(identity_cert_name);
+  }
+
  private:
   grpc_ssl_client_certificate_request_type cert_request_type_;
   grpc_tls_server_verification_option server_verification_option_ =
       GRPC_TLS_SERVER_VERIFICATION;
   grpc_tls_version min_tls_version_ = grpc_tls_version::TLS1_2;
   grpc_tls_version max_tls_version_ = grpc_tls_version::TLS1_3;
-  grpc_core::RefCountedPtr<grpc_tls_key_materials_config> key_materials_config_;
-  grpc_core::RefCountedPtr<grpc_tls_credential_reload_config>
-      credential_reload_config_;
   grpc_core::RefCountedPtr<grpc_tls_server_authorization_check_config>
       server_authorization_check_config_;
+  grpc_tls_certificate_provider* grpc_tls_certificate_provider_;
```

Comment:
I am not entirely sure whether `grpc_tls_credentials_options` should take a RefCountedPtr or a raw pointer of  grpc_tls_certificate_provider. I used to use RefCountedPtr, but that caused some problems when I tried to define
```
int grpc_tls_credentials_options_set_certificate_provider(grpc_tls_credentials_options* options,
    grpc_tls_certificate_provider* provider)
```
in `grpc_security.h`. If options takes a RefCountedPtr of grpc_tls_certificate_provider, this function will take in a raw pointer and convert it to a RefCountedPtr, which implicitly grab the whole ownership, since the caller of C-core APIs doesn't have a way to get the RefCountedPtr out of the options. 
Letting options to control provider is not what we want here. We want the provider to outlive the options, and hence I am using a raw pointer. 
I am not sure if I missed anything, or if the above reasoning is correct. @markdroth Can you please take a look? Feel free to let me know if the question doesn't make sense, and I can add more details. Thanks in advance!